tensorflow/core/profiler/internal/gpu/cupti_tracer.cc

```diff
@@ -1380,32 +1381,47 @@ void CuptiTracer::Enable(const CuptiTracerOptions &option,
   }
 
   EnableApiTracing().IgnoreError();
-  if (option_->enable_activity_api) {
+  if (!subscriber_existed_ && option_->enable_activity_api) {
     EnableActivityTracing().IgnoreError();
   }
 }
 
 void CuptiTracer::Disable() {
-  DisableApiTracing().IgnoreError();
-  if (option_->enable_activity_api) {
-    DisableActivityTracing().IgnoreError();
+  if (!subscriber_existed_) {
+    DisableApiTracing().IgnoreError();
+    if (option_->enable_activity_api) {
+      DisableActivityTracing().IgnoreError();
+    }
+    cupti_interface_->CleanUp();
+    Finalize().IgnoreError();
+    cupti_driver_api_hook_->Flush().IgnoreError();
+    collector_->Flush();
+    collector_ = nullptr;
+    option_.reset();
+    cupti_driver_api_hook_.reset();
   }
-  cupti_interface_->CleanUp();
-  Finalize().IgnoreError();
-  cupti_driver_api_hook_->Flush().IgnoreError();
-  collector_->Flush();
-  collector_ = nullptr;
-  option_.reset();
-  cupti_driver_api_hook_.reset();
 }
 
 Status CuptiTracer::EnableApiTracing() {
   if (api_tracing_enabled_) return Status::OK();
   api_tracing_enabled_ = true;
 
   VLOG(1) << "Enable subscriber";
-  RETURN_IF_CUPTI_ERROR(cupti_interface_->Subscribe(
-      &subscriber_, (CUpti_CallbackFunc)ApiCallback, this));
+  CUptiResult status = cupti_interface_->Subscribe(
+      &subscriber_, (CUpti_CallbackFunc)ApiCallback, this);
+  if (status != CUPTI_SUCCESS) {
+    const char *errstr = "";
+    cupti_interface_->GetResultString(status, &errstr);
+    if (status == CUPTI_ERROR_MAX_LIMIT_REACHED) {
+      subscriber_existed_ = true;
+      LOG(WARNING) << "function cupti_interface_->Subscribe failed with error "
+                   << errstr << " and TF CUPTI tracing will be ignored.\n";
+    } else {
+      LOG(ERROR) << "function cupti_interface_->Subscribe failed with error "
+                 << errstr;
+    }
+    return errors::Internal(absl::StrCat("cupti call error", errstr));
```

Comment:
I believe this can just be `errors::Internal("cupti call error", errstr)`.

Also, should there be a space after `"cupti call error"`?